_pytest/junitxml.py

```diff
@@ -53,6 +53,13 @@ def repl(matchobj):
             return unicode('#x%04X') % i
     return py.xml.raw(illegal_xml_re.sub(repl, py.xml.escape(arg)))
 
+def record_property(name, value):
+    if hasattr(record_property, 'binding'):
+        record_property.binding(name, value)
+
+def pytest_namespace():
```

Comment:
Instead of a function, perhaps this would be easier to implement and understand if you implement `record_property` (or `record_xml_property`) as a fixture? That way you have access to the `_xml` object from within it:

```python
@pytest.fixture
def record_xml_property(request):
    return request.config._xml.record_property
```

Then usage stays very similar:

```python
def test_record(record_xml_property):
    record_xml_property("foo", "<1");
```